### ISSUE TITLE
clear old data

### ISSUE THREAD

**Issue by Vikinge** [nibi79/synologysurveillancestation#39]
I am replacing my docker image for 6th time, dragging along the conf,userdata and addons folders.  I removed the .jar file from addons, but after restart and re-add of addon I cannot create binding, error 409-conflict.  Is there an easy or hard way to clear out all things/references to synologysurveillance and start over.  My initial install of addon worked great.

**Comment by nibi79** [comment]
Did you try to clear the cache?
https://community.openhab.org/t/clear-the-cache/36424

**Comment by b-r-y** [comment]
Error 409 in the API says password expired. Could that be another direction to follow?

**Comment by Vikinge** [comment]
I will try clearing the cache and tmp, I had not done that before.  I am
using my Synology admin user, but I see surveillance has it's own user
accounts as well, do you create a dedicated account there with limited
access?

**Comment by b-r-y** [comment]
Yes you can create a limited account as viewer only. Most features do not require admin ( pan, zoom, snapshot, record will all work with viewer account). Turning on/off motion control ( and other changes related to MC) does require admin but this can be SS only admin and does not have to be DS admin.

**Comment by Vikinge** [comment]
thanks.  purging temp and cache and re-adding .jar did not help, same
409-conflict.  I know there are things defined for surveillance, is that
causing the conflict, or is there some piece of the binding that is
preventing me from re-adding?

On a different note, my first couple docker builds things were going great,
but I was creating persistent storage wrong.  Now I am mounting the three
folders and not an entire /openhab folder.. However, on every reboot, I
lose all configured things.  My inbox is filled with newly identified
entities on boot and each one has to be re-defined again.  I assume this is
not normal, correct?

**Comment by Pavion** [comment]
Hi Viki and hello together!

Please check in your Karaf console, whether the binding is installed: 
```
openhab> bundle:list | grep Syno
275 x Active    x  80 x 2.5.0.a37              x SynologySurveillanceStation Binding
```
That is the output for an active and running binding. If you encounter any issues, please remove the JAR from /addons and check binding's state again. If it's still there, please execute 
```
bundle:uninstall 275 
``` 
where 275 is the number of the binding shown above.
Just in case, restart your openHAB using `system:shutdown` and try to reinstall the binding after that. 

Some info follows.

**Comment by Pavion** [comment]
@nibi79 @b-r-y 
Please note, error 409 is most likely **not** API related but an HTTP error:
https://httpstatuses.com/409

You will see this error if you create an Thing using .things file and then try to delete it from PaperUI. This is an invalid action and is signalized by ESH as ERROR 409:

![image](https://user-images.githubusercontent.com/1370084/56295581-4b85e700-612d-11e9-81f2-e82f6967ecc1.png)

My assumption on this is, that some folders of @Vikinge's installation are read-only for his docker user, so no changes made in PaperUI are persisted. @Vikinge please check this.

**Comment by b-r-y** [comment]
No disappearing configs means your docker overwrites things rather than using your folders.

This is my run command:

docker run \
      --name openhab \
      --net=host \
      --tty \
      -e “EXTRA_JAVA_OPTS=-Dgnu.io.rxtx.SerialPorts=-Dgnu.io.rxtx.SerialPorts=/dev/ttyUSB0:/dev/ttyS0:/dev/ttyUSB0:/dev/ttyS2:/dev/ttyACM0:/dev/ttyAMA \
      -v /volume1/docker/openhab/conf:/openhab/conf \
      -v /volume1/docker/openhab/userdata:/openhab/userdata \
      -v /volume1/docker/openhab/addons:/openhab/addons \
      -d \
      -e OPENHAB_HTTP_PORT=8083 \
      -e OPENHAB_HTTPS_PORT=8443 \
      --device=/dev/ttyACM0 \
	  --device=/dev/ttyUSB0 \
      --restart=always \
      openhab/openhab:2.4.0-amd64-debian

You can see the three folders and how they are defined. This uses docker 1 syntax as this is what I have but should work on newer as well.

**Comment by Vikinge** [comment]
I have my volumes defined exactly as you listed.  I just forced all
permissions to full rw (I assume 777 is what synology set).  I will let you
know.

**Comment by Vikinge** [comment]
> @nibi79 @b-r-y
> Please note, error 409 is most likely **not** API related but an HTTP error:
> https://httpstatuses.com/409
> 
> You will see this error if you create an Thing using .things file and then try to delete it from PaperUI. This is an invalid action and is signalized by ESH as ERROR 409:
> 
> ![image](https://user-images.githubusercontent.com/1370084/56295581-4b85e700-612d-11e9-81f2-e82f6967ecc1.png)
> 
> My assumption on this is, that some folders of @Vikinge's installation are read-only for his docker user, so no changes made in PaperUI are persisted. @Vikinge please check this.

I did touch the .things file, but no entries were made.  I just tried removing .jar, removing binding with bundle:uninstall, restart, added jar, click bindings, added user/password, left with http and verified ip and port.  Still fails 209-Conflict.  The status shows uninitialized - handler_configuration_pending

<img width="1008" alt="Screen Shot 2019-04-17 at 10 30 26 AM" src="https://user-images.githubusercontent.com/725760/56296396-69872380-60fc-11e9-82dc-1ba8ac258f7f.png">

**Comment by Pavion** [comment]
I'm not using docker myself and so can't help further. If you're getting an error 409 as shown above, it's not related to Synology API but is an common error. If other bindings don't work proper as well, you are having some common issue with your file permissions. 
If you are using text file configuration, your `conf/items` and `conf/things` should be readable at least.
If you are using PaperUI only, your `userdata/jsondb` should be writtable. 
@Vikinge please check/confirm, whether your issue is binding related or a common one. I will still try my best to help you but I have a feeling we're discussing different things all along.

**Comment by Vikinge** [comment]
I am pretty sure its docker related, but have not found it yet.  I added hue two days in build 5 and it added just fine.  Today I just added darksky, after initializing for a while it showed 409-conflict.  I think you are onto something with permissions.  I will keep going down that path for now.  I appreciate your help, I can probably find the answer from here.  

Can I manually edit the jsondb without breaking things (if I correct edit json data)?  
I read that when upgrading to a new host (docker image) you copy the conf,addons and userdata, is this true?  Could I delete the jsondb and retain all my services,bindings?

**Comment by Pavion** [comment]
/jsondb is where **PaperUI** stores all its Things, Items and their connections. Deleting this folder will thus delete all of this. You may also edit these files (**backup them first!**) to remove non-existent items: there **are** cases when an item is kept upon deletion and have to be removed manually. 

For example, if you remove a Thing you've created in PaperUI and **simultaneously** create the same Thing using text file, jsondb record is **not** deleted, can provoke an conflict and must be removed manually. 

Generally, if you're using text files only, your jsondb files **should** be empty. There are but few bindings, which **require** PaperUI usage (e.g. ZWave Binding).

**Comment by b-r-y** [comment]
I now also think permissions on the docker looks most likely. The openhab container starts by default with user openhab. I made such user in DS but never managed to get it to run properly. When running docker as I showed above it actually runs the docker comand as root. Yet the container runs as openhab. In my case this gives uid 9001 for all created files. I could never match that to a DS user openhab with the same uid. But if you change the files you want to use in those folders and make the owner 9001 ( or whatever it comes out on your end) it should work. At least it does for me.

**Comment by Vikinge** [comment]
Sooo.  I changed permissions from synology (docker host) to allow everyone read/write.  before I assumed docker was running as me and gave my user full ownership, forced full read/write permission on all folders and subdir.  I added plex binding, no error.  Synology surveillance addon is still in the addons folder, so I chose things/+/synology surveillance and clicked on the one entry.  Now I see error 404 not found.  Do I restart all my diagnostics at the beginning since the 409 error was probably permissions?  Not sure what it is doing that would results in not found?
